### ISSUE TITLE
2 rpc-like methods have very low performance issues

### ISSUE THREAD

**Issue by sundafa** [ledgerwatch/erigon#5911]
#### System information

Erigon version: `./erigon --version`
erigon version 2.28.1-stable-9711af6d

OS & Version: Windows/Linux/OSX
ubuntu 20.04
chain: eth
Hetzenr: [AX101](https://www.hetzner.com/dedicated-rootserver/ax101)
CPU: AMD Ryzen™ 9 5950X 

Commit hash : 

#### Expected behaviour
eth_gasPrice
eth_estimateGas
The performance of these two methods is similar to the geth client when using the request

#### Actual behaviour
Test out qps performance comparison:
erigon
eth_gasPrice  4k/s
eth_estimateGas  19k/s

geth
eth_gasPrice  92k/s
eth_estimateGas  57k/s

erigon eth_gasPrice : k6 tool results
![image](https://user-images.githubusercontent.com/24621599/198942352-8ed167f6-4d17-4421-9362-a6c831850ba1.png)

geth eth_gasPrice : k6 tool results
![image](https://user-images.githubusercontent.com/24621599/198942364-9705f35d-a4ff-42a0-a452-4b862c4df6f6.png)

#### Steps to reproduce the behaviour
Use the stress test tool to perform the same rpc method stress test

#### Backtrace

````
[backtrace]
````

**Comment by AskAlexSharov** [comment]
@sundafa , hi. 
what if you increase Erigon's --db.read.concurrency to 1K or 5K?

**Comment by sundafa** [comment]
The qps of other interfaces are all at 80k~90k, only these two interfaces are abnormal

**Comment by hexoscott** [comment]
Hi @sundafa, I'm going to take a look into this.  Could you send me some example calls that you're using in your tests please?  Thanks!

**Comment by sundafa** [comment]
@hexoscott 
This is the k6 script, simplified
After modifying the ip and port, run the k6 command `k6 run -u3000 -d30s name.js`
name.js
```
import http from 'k6/http';
import { check } from 'k6';

let url = 'http://IP:PORT';

export default function () {
    eth_gasPrice()
    eth_estimateGas()
}

function checkErr(res) {
    check(res, {
        "status_code_200": (r) => r.status === 200,
        "status_code_404": (r) => r.status === 404,
    })
}

let params = {
    headers: {
        'Content-Type': 'application/json',
    },
};

function eth_gasPrice() {
    const payload = JSON.stringify({ 'jsonrpc': '2.0', 'method': 'eth_gasPrice', 'params': [], 'id': 21 });
    const res = http.post(url, payload, params);
    checkErr(res)
}
function eth_estimateGas() {
    const payload = JSON.stringify({ 'jsonrpc': '2.0', 'method': 'eth_estimateGas', 'params': [], 'id': 21 });
    const res = http.post(url, payload, params);
    checkErr(res)
}
```

**Comment by hexoscott** [comment]
Great thanks, will take a look

**Comment by sundafa** [comment]
You can use these 2 methods individually, just test one at a time
```
    eth_gasPrice()
   #  eth_estimateGas()
```

**Comment by AskAlexSharov** [comment]
@hexoscott FYI probably there can use BlocksLRU instead of reading header from db. Because this methods need only recent blocks

**Comment by hexoscott** [comment]
Great advice, thanks!

**Comment by hexoscott** [comment]
Two PRs to handle the performance of these calls.  They make some lightweight changes and re-introduce caching for eth_gasPrice to help with performance.  We can take a deeper dive if performance is still an issue here.  

The eth_estimateGas changes are merged, once #6016 is merged these two calls should be more performant.

**Comment by sundafa** [comment]
I tested it just now, and now the rqs has not improved much under the v2.29.0 version. I will continue to pay attention to the next version.

**Comment by sundafa** [comment]
erigon version: v2.30.0
After testing after updating the node just now
The performance of the RPC method `eth_syncing` is greatly improved, the CPU only uses 50% performance, from 4k/s -> 86k/s
The performance of the RPC method `eth_estimateGas` is not much improved, the cpu uses 100% performance, from 19k/s -> 24k/s

Thanks a lot for your help

**Comment by hexoscott** [comment]
Hi @sundafa, thanks for the feedback, how do those stats look with `--db.read.concurrency=5000`?

**Comment by hexoscott** [comment]
Hi @sundafa, some more recent changes have been merged into devel branch if you wanted to check them out.  They should improve gas estimation performance

**Comment by sundafa** [comment]
ok, i'll try that in my free time
